be/src/storage/rowset/column_writer.cpp

```diff
@@ -898,7 +898,7 @@ StringColumnWriter::StringColumnWriter(const ColumnWriterOptions& opts, std::uni
         : ColumnWriter(std::move(field), opts.meta->is_nullable()), _scalar_column_writer(std::move(column_writer)) {}
 
 Status StringColumnWriter::append(const vectorized::Column& column) {
-    DCHECK(check_string_lengths(column).ok());
+    RETURN_IF_ERROR(check_string_lengths(column));
```

Comment:
why enable it? This is just for test.

Reply:
In some user scenarios, it is observed that some string's length exceed the upper limit of varchar

Reply:
OK I think this is a defensive programing.
This will make users load fail.
Could you test to see the performance degrade which is brought by this check?

Reply:
It seems has no impact on performance.
test with `test_string_column_writer_benchmark`,
Without check: 20907ms
With check: 20918ms

Reply:
Cool